pkg/middleware/instrumentation.go

```diff
@@ -0,0 +1,120 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/justinas/alice"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func NewRequestInstrumentorWithDefaultRegistry(metricsPath string) alice.Constructor {
+	return NewRequestInstrumentor(prometheus.DefaultRegisterer, metricsPath)
+}
+
+func NewRequestInstrumentor(registerer prometheus.Registerer,
+	metricsPath string) alice.Constructor {
+	return func(next http.Handler) http.Handler {
+
+		// Counter for all requests
+		// This is bucketed based on the response code we set
+		requestsCounter := RegisterRequestsCounter(registerer)
+
+		// Gauge to all requests currently being handled
+		inFlightRequests := RegisterInflightRequestsGauge(registerer)
+
+		// The latency of all requests bucketed by HTTP method
+		requestLatency := RegisterRequestsLatencyHistogram(registerer)
```

Comment:
Yeah, it was mostly for testing.

Reply:
It may come in useful in the future. Using the global registry all sorts of other packages will add their own metrics, that may or may not be desirable so we may want to use a custom registry